test/Microsoft.AspNet.Mvc.FunctionalTests/ModelMetadataAttributeTest.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.TestHost;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.FunctionalTests
+{
+    public class ModelMetadataAttributeTest
+    {
+        private readonly IServiceProvider _services = TestHelper.CreateServices(nameof(ValidationWebSite));
+        private readonly Action<IApplicationBuilder> _app = new ValidationWebSite.Startup().Configure;
+
+        [Fact]
+        public async Task ValidBaseClass_Product()
```

Comment:
rename all test methods to include expectations; follow the `method_conditions_expectations` or `method_expectations_conditions` pattern (preferably, pick one :smiley_cat:)